app/rust/src/parser/plans/output.rs

```diff
@@ -61,13 +69,48 @@ impl OutputPlanC {
         let body = self.body(fvk, &memo_payload_key);
 
         if let Ok(body) = body {
-            let mut state =
-                create_personalized_state("/penumbra.core.component.shielded_pool.v1.OutputBody");
-
-            state.update(&body.note_payload.to_proto());
-            state.update(&body.balance_commitment.to_proto_output());
-            state.update(&body.wrapped_memo_key.to_proto());
-            state.update(&body.ovk_wrapped_key.to_proto());
+            let mut state = create_personalized_state(
+                std::str::from_utf8(OUTPUT_PERSONALIZED)
+                    .expect("OUTPUT_PERSONALIZED must be valid UTF-8"),
```

Comment:
```suggestion
                    .map_err(|_| ParserError::UnexpectedError)?
```
std::str::from_utf8(OUTPUT_PERSONALIZED)

Reply:
maybe add an `InvalidUtf8` error variant to our error definition in error.rs